In [1]:
# Imports
import os
import sys
import cmath
import math
import matplotlib.pyplot as plt
import matplotlib
import numpy as np
import pandas as pd
import sympy as sp
from sympy.utilities.lambdify import lambdify
from scipy import signal
%matplotlib inline
from IPython.core.interactiveshell import InteractiveShell
InteractiveShell.ast_node_interactivity = "all"
from matplotlib.ticker import LogLocator

In [2]:
def par(R1,R2):
    return (1/R1 + 1/R2)**-1

**Problem 1**

$\begin{eqnarray}
\mathcal{R}_1 &=& \frac{\ell _c}{\mu_0 \mu_r A_{c_1}} \\[0.5em]
&=& \frac{3\cdot 10^{-2}\left[m\right]}{(4\pi \cdot 10^{-7})\left[\frac{H}{m}\right] (1000) (10^{-2} \cdot 10^{-2})\left[m^2\right]} \\[0.5em]
&=& 2.39\cdot 10^5 H^{-1}
\end{eqnarray}$

$\begin{eqnarray}
\mathcal{R}_2 &=& \frac{\ell _c}{\mu_0 \mu_r A_{c_2}} \\[0.5em]
&=& \frac{3\cdot 10^{-2}\left[m\right]}{(4\pi \cdot 10^{-7})\left[\frac{H}{m}\right] (1000) (10^{-2} \cdot 0.5\cdot10^{-2})\left[m^2\right]} \\[0.5em]
&=& 4.77\cdot 10^5 H^{-1}
\end{eqnarray}$

In [3]:
lc = 3E-2
mu0 = 4*np.pi*1E-7
mur = 1000
Ac = 1E-2*1E-2

In [4]:
R1 = lc/(mu0*mur*Ac)
R1

238732.414637843

In [5]:
R2 = lc/(mu0*mur*0.5*Ac)
R2

477464.829275686

In [6]:
Req = 3*R1 + par(R1,2*R1 + R2)
Req

907183.1756238034

**Problem 2**

In [7]:
10**2/Req

0.00011023132117858924

$\begin{eqnarray}
L &=& \frac{n^2}{\mathcal{R}_c + \mathcal{R}_g} = \frac{n^2}{\mathcal{R}_{eq}} \\[0.5em]
\mathcal{R}_{eq} &=& 3R_2 + \left[ R_1\parallel (2R_1+R_2) \right] =  907183 \ H^{-1}\\[0.5em]
&=&  \frac{10^2}{90.7\cdot10^5 \ [H^{-1}]} \\[0.5em]
&=& 110 \mu H
\end{eqnarray}$

**Problem 3**

**Problem 4**

<hr>

<img src="img/TwoWindingTransformer.png" width="600">

<hr>

In [8]:
Req2 = R2 + 2*R1 + par(3*R1,R1)
Req2

1133978.9695297542

In [28]:
R1/(3*R1 + R2)

0.19999999999999998

In [9]:
(10**2)/Req

0.00011023132117858924

In [30]:
0.25*(10*20)/Req2

4.40925284714357e-05

In [31]:
0.2*(10*20)/Req

4.40925284714357e-05

$\begin{eqnarray}
\Phi &=& \Phi_1 + \Phi_2 \\[0.5em]
\Phi_1 &=& \Phi_{11} [\text{flux generated from source 1}]  \\[0.5em] 
&+&\Phi_{12} [\text{flux in winding 1 due to current from winding 2}] \\[0.5em]
\Phi_2 &=& \Phi_{22} [\text{flux generated from source 2}] \\[0.5em] 
&+& \Phi_{21} [\text{flux in winding 2 due to current from winding 1}] \\[0.5em]
\end{eqnarray}$

$\begin{eqnarray}
\mathcal{R}_{eq_1} &=& 3R_2 + \left[ R_1\parallel (2R_1+R_2) \right] \\[0.5em]
&=& 907183 \ H^{-1} \ [\text{source 1 only}] \\[0.5em]
\mathcal{R}_{eq_2} &=& 2R_1 + R_2 + \left[ 3R_1\parallel R_1 \right] \\[0.5em]
&=&  1133979 \ H^{-1}\ [\text{source 2 only}] \\[0.5em]
\end{eqnarray}$

<hr>

**Turn off source 2**

$\begin{eqnarray}
\Phi_{11} &=& \frac{n_1 i_1}{\mathcal{R}_{eq_1}} \\[0.5em]
\Phi_{21} &=& \Phi_{11} \cdot \frac{R_1}{R_1 + (2R_1 + R_2)} = 0.20\ \Phi_{11} \\[0.5em]
\end{eqnarray}$

<hr>

**Turn off source 1**

$\begin{eqnarray}
\Phi_{22} &=& \frac{n_2 i_2}{\mathcal{R}_{eq_2}} \\[0.5em]
\Phi_{12} &=& \Phi_{22} \cdot \frac{R_1}{R_1 + 3R_1} = 0.25\ \Phi_{22}\\[0.5em]
\end{eqnarray}$

<hr>

$\begin{eqnarray}
v_1 &=& n_1\frac{d}{dt}\left( \Phi_{1} \right) = n_1\frac{d}{dt}\left( \Phi_{11}+\Phi_{12} \right)\\[0.5em]
v_1 &=& n_1\frac{d}{dt}\left( \Phi_{11}\right) +  n_1\frac{d}{dt}\left(\Phi_{12} \right) \\[0.5em]
v_1 &=& n_1\frac{d}{dt}\left(\frac{n_1 i_1}{\mathcal{R}_{eq_1}}\right) + n_1\frac{d}{dt}\left(0.25 \frac{n_2 i_2}{\mathcal{R}_{eq_2}}\right)\\[0.5em]
v_1 &=& \frac{n_1^2}{\mathcal{R}_{eq_1}}\frac{d}{dt}(i_1) + 0.25\frac{n_1 n_2}{\mathcal{R}_{eq_2}}\frac{d}{dt}(i_2) \\[0.5em]
v_1 &=& L_{11}\frac{d}{dt}(i_1) + L_{12}\frac{d}{dt}(i_2) \\[0.5em]
L_{11} &=& \frac{n_1^2}{\mathcal{R}_{eq_1}} \\[0.5em]
L_{11} &=& \frac{10^2}{907183} = 110 \mu H \\[0.5em]
L_{12} &=& 0.25\frac{n_1 n_2}{\mathcal{R}_{eq_2}} \\[0.5em]
L_{12} &=& 0.25 \left(\frac{10\cdot 20}{1133979}\right) = 44 \mu H \\[0.5em]
\end{eqnarray}$

<hr>

$\begin{eqnarray}
v_2 &=& n_2\frac{d}{dt}\left( \Phi_{2} \right) = n_2\frac{d}{dt}\left( \Phi_{21}+\Phi_{22} \right)\\[0.5em]
v_2 &=& n_2\frac{d}{dt}\left( \Phi_{21}\right) +  n_2\frac{d}{dt}\left(\Phi_{22} \right) \\[0.5em]
v_2 &=& n_2\frac{d}{dt}\left(0.2 \frac{n_1 i_1}{\mathcal{R}_{eq_1}}\right) + n_2\frac{d}{dt}\left(\frac{n_2 i_2}{\mathcal{R}_{eq_2}}\right)\\[0.5em]
v_2 &=& 0.2\frac{n_1 n_2}{\mathcal{R}_{eq_1}}\frac{d}{dt}(i_1) + \frac{n_2^2}{\mathcal{R}_{eq_2}}\frac{d}{dt}(i_2)\\[0.5em]
v_2 &=& L_{12}\frac{d}{dt}(i_1) + L_{22}\frac{d}{dt}(i_2) \\[0.5em]
L_{12} &=& 0.2\frac{n_1 n_2}{\mathcal{R}_{eq_1}} \\[0.5em]
L_{12} &=& 0.2 \left(\frac{10\cdot 20}{907183}\right) = 44 \mu H \\[0.5em]
L_{22} &=& \frac{n_2^2}{\mathcal{R}_{eq_2}} \\[0.5em]
L_{22} &=& \frac{20^2}{1133979} = 352 \mu H \\[0.5em]
\end{eqnarray}$

<hr>